### PR TITLE
warnet: generate as files

### PR DESCRIPTION
**Pull request by willcl-ark**
generates ASmap files and `COPY`s them into the containers

### CODE REVIEW

**Comment by willcl-ark** [comment]
@pinheadmz @josibake 

If y'all want ASmap, then we ready for review in here :)

File: src/warnet/warnet.py [hunk 1]
```diff
@@ -38,7 +41,9 @@ def __init__(self, config_dir):
         self.graph: Optional[networkx.Graph] = None
         self.graph_name = "graph.graphml"
         self.tanks: List[Tank] = []
-
+        self.a_systems = set()
+        self.as_map_txt_path = config_dir / "asmap.txt"
+        self.as_map_dat_path = config_dir / "asmap.dat"
```
Comment (josibake) [inline src/warnet/warnet.py]: should probably get in the habit of not hardcoding files names. in the past, i think this has made it hard for us to pull things out / refactor

File: src/warnet/warnet.py [hunk 2]
```diff
@@ -122,6 +127,43 @@ def tanks_from_graph(self):
                 )
             self.tanks.append(Tank.from_graph_node(node_id, self))
         logger.info(f"Imported {len(self.tanks)} tanks from graph")
+        self.generate_as_map()
+
+    def generate_as_map(self):
+        # Write AS mappings to file
+        with open(self.as_map_txt_path, "w") as f:
+            for tank in self.tanks:
+                f.write(f"{tank.ipv4}/32 AS{tank.autonomous_system}\n")
```
Comment (josibake) [inline src/warnet/warnet.py:141]: why "/32" here? will this /32 work regardless of where warnet is running?

File: src/warnet/warnet.py [hunk 3]
```diff
@@ -122,6 +127,43 @@ def tanks_from_graph(self):
...
+        logger.info(f"AS map: Total bits: {len(ser)}")
+        with open(self.as_map_dat_path, "wb") as f:
+            f.write(bytes(buildmap.EncodeBytes(ser)))
+
+        # Reset sys.stdin to its original state
+        sys.stdin = sys.__stdin__
+
+        # load the file into each bitcoin node
+
+        # have each node read it on startup
```
Comment (josibake) [inline src/warnet/warnet.py]: not finished yet? or are these comments outdated?

**Review by josibake — COMMENTED** [review]
left a few comments

File: src/warnet/warnet.py (hunk 2 above)
Comment (willcl-ark) [reply src/warnet/warnet.py:141]: That's means "a single ip address"

File: src/warnet/warnet.py (hunk 1 above)
Comment (willcl-ark) [reply src/warnet/warnet.py]: Oh, thought I took that out like I did in the remove volumes fixup

File: src/warnet/warnet.py (hunk 3 above)
Comment (willcl-ark) [reply src/warnet/warnet.py]: Hmmm, the combo of this and the first comment makes me think my rebase went wrong. 

Will address after bathtime 🐤

Comment (willcl-ark) [reply src/warnet/warnet.py]: OK addressed in 08f4dc15bb71c6b4b3b646bb5d41bfd916f628ed

File: src/warnet/warnet.py (hunk 1 above)
Comment (willcl-ark) [reply src/warnet/warnet.py]: fixed in 08f4dc15bb71c6b4b3b646bb5d41bfd916f628ed

**Comment by willcl-ark** [comment]
Closing for #217

**Pull request closed without merging**